Jetpack App (Emphasis): Remove "Followed Sites" from notifications settings

ashiagr opened the pull request:
Fixes #14512 

This PR removes the "Followed Sites" section from the notifications settings for the Jetpack app. 

To test:

1. Install the Jetpack app.
2. Go to the "Notifications" tab.
3. Click the gear icon on the top right corner to go to the "Notifications Settings" screen.
4. Notice that the followed sites are not displayed on the screen. 

Jetpack | WordPress
---------|-----------
![notifs_settings](https://user-images.githubusercontent.com/1405144/120068444-58ac8880-c09e-11eb-8e36-171188c3ec16.png) | ![notifs_settings_wp](https://user-images.githubusercontent.com/1405144/120069854-ad9fcd00-c0a5-11eb-9136-7d7df24c3de3.png)

## Regression Notes
1. Potential unintended areas of impact 
 WordPress Notification Settings

2. What I did to test those areas of impact (or what existing automated tests I relied on) 
I tested the WordPress app and made sure the Followed Sites section is displayed

3. What automated tests I added (or what prevented me from doing so) 
N/A

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

ashiagr committed:
Remove the "Followed Sites" section from the notifications settings for the Jetpack app

ParaskP7 commented on WordPress/src/main/java/org/wordpress/android/ui/prefs/notifications/NotificationsSettingsFragment.java:
```diff
@@ -126,6 +128,7 @@ public void onCreate(Bundle savedInstanceState) {
         addPreferencesFromResource(R.xml.notifications_settings);
         setHasOptionsMenu(true);
         removeSightAndSoundsForAPI26();
+        if (mBuildConfigWrapper.isJetpackApp()) removeFollowedBlogsPreference();
```
Suggestion (💡): I would have considered keeping this consistent with the above method, calling it a `removeFollowedBlogsForJetpack()` instead and including the `if/else` logic within, just like it happens above. This is purely for consistency purposes, maybe slightly for readability, but not sure. Feel free to ignore if you feel otherwise or you find it super minor.

ashiagr replied:
I was actually toying with this idea, and I then reverted to the present form as calling the method for the WordPress app was not required and we pre-checked it by putting the if check.  I will go ahead and move the condition inside the method as it came to my mind as well, and it is brought up in this discussion.

In commit 71117b5

ParaskP7 replied:
Awesome, many thanks Ashita! 🚀

ParaskP7 commented on WordPress/src/main/java/org/wordpress/android/ui/prefs/notifications/NotificationsSettingsFragment.java:
```diff
@@ -567,7 +578,7 @@ private void configureBlogsSettings(PreferenceCategory blogsCategory, boolean sh
 
     private void configureFollowedBlogsSettings(PreferenceCategory blogsCategory, final boolean showAll) {
-        if (!isAdded()) {
+        if (!isAdded() || blogsCategory == null) {
```
1) Suggestion (💡): It wasn't obvious to me why this change happened and as such I actually played a bit with it to understand more about it. Ideally, I would have either made this change a separate commit or I would have added a commit description on why this extra bit of logic was added here (ie. null pointer exception because of this and that).
2) Suggestion (💡): I would consider adding `@Nullable` annotations everywhere related to this change to make it more obvious what is what. For example, I would have added the `@Nullable` annotation on the `blogsCategory` parameter, just so that it becomes more evident that the `blogsCategory.removeAll();` line might produce null pointer exception. Then, I would have also considered to add the `@Nullable` annotation on the `mFollowedBlogsCategory` field to indicate that this can be nullable going forward, just to avoid any future surprises with it. Wdyt?

ashiagr replied:
Great suggestions! 

`@Nullable` annotation added in 5fa620e.

ParaskP7 reviewed (approved):
👋 @ashiagr !

I have reviewed and tested this PR as per the instructions, everything works as expected, good job! 🌟 

I have left some suggestions (💡) for you to consider. I am going to approve this PR anyway, since none is blocking. I am **NOT** going to merge this PR yet to give you some time to apply any of my suggestions. However, feel free to ignore them and merge the PR yourself.

ashiagr committed:
Rename method and move if Jetpack app condition inside it

ashiagr committed:
Add nullable annotation for the followed blog pref category

Since the followed blog pref category can be null for the Jetpack app, it is annotated with @Nullable to make it more evident.

ParaskP7 reviewed (approved):
👋 @ashiagr !

Thanks so much for applying my suggestions, all LGTM, feel free to merge this PR when CI completes! 🚀 ❤️
